Stellar-contract.x

```diff
@@ -270,27 +226,81 @@ case SC_ADDRESS_TYPE_CONTRACT:
     Hash contractId;
 };
 
-union SCObject switch (SCObjectType type)
+%struct SCVal;
+%struct SCMapEntry;
+
+const SCVAL_LIMIT = 256000;
+const SCSYMBOL_LIMIT = 256;
+
+typedef SCVal SCVec<SCVAL_LIMIT>;
+typedef SCMapEntry SCMap<SCVAL_LIMIT>;
+
+typedef string SCString<SCVAL_LIMIT>;
+typedef string SCSymbol<SCSYMBOL_LIMIT>;
+
+union SCVal switch (SCValType type)
 {
-case SCO_VEC:
-    SCVec vec;
-case SCO_MAP:
-    SCMap map;
-case SCO_U64:
+
+case SCV_BOOL:
+    bool b;
+case SCV_VOID:
+    void;
+case SCV_STATUS:
+    SCStatus error;
```

Comment:
Can we name this variable status to reduce the concept/language space? Alternatively, rename SCStatus to SCError?
```suggestion
    SCStatus status;
```

Reply:
I think we'd benefit from renaming Status to Error. We use the term "Error" with developers in docs.

Reply:
+1 for Error, I have an expectation for status to have a positive variant (like Status::Ok/Success), while IIUC we have only errors here.

Reply:
I don't think we use the positive form of status anywhere, and we should just focus on this being an error instead.

Reply:
`contract_event` recording returns `ScStatus::Ok` if the the event was successfully recorded. https://github.com/stellar/rs-soroban-env/blob/main/soroban-env-common/env.json#L56

Reply:
It looks like in the code there's very little use of `Ok`. We could change any cases where we are returning Ok to returning Void, and get rid of Ok.

Reply:
I actually made this change earlier and then changed it back for two reasons:

  - There are a bunch of other renamings I want to do that I thought we could do a "semantic no-op, just renaming" PR and bundle them all together.
  - This _particular_ renaming introduces a ton of churn because `Error` is a type-name used ubiquitously in Rust code already (as an associated type of lots of traits)

On the other most of the renaming I had in mind is env/sdk-side rather than XDR-side, and if you want to minimize the number of XDR PRs I'm willing to try including the Status=>Error change in this one. I totally agree it should happen eventually.

Reply:
👍 let's do it separately.

Reply:
will hold on Status => Error for now (added to tasklist on #679)